changelogs/current.yaml

```diff
@@ -94,6 +94,10 @@ minor_behavior_changes:
   change: |
     Added new type of gauge with type hidden. These stats are hidden from admin/stats-sinks but can shown with a
     query-parameter of ``/stats?hidden=include`` or ``/stats?hidden=showonly``.
+- area: upstream
+  change: |
+    Changed behavior of the request encoder related to half closed connections. This change can be reverted temporarily by
```

Comment:
how about
changed the HTTP/1 codec to not delay sending a half close on HTTP end stream any time the connection has half close enabled

Reply:
> awesome!
> 
> /wait on CI - one of the test failures looks real

yeah. 
half-close is not enabled on the upstream side connection. It is [set only on the downstream side connection](https://github.com/envoyproxy/envoy/blob/main/source/common/tcp_proxy/tcp_proxy.cc#L237).   upstream side connection, [TcpProxy::HttpUpstream calls enableTcpTunneling](https://github.com/envoyproxy/envoy/blob/main/source/common/tcp_proxy/upstream.cc#L324) on request_encoder_ because it does not have access to upstream connection

So I think change that I am doing in another PR to set enable_tcp_tunneling_ in UpstreamRequest will be required because with UpstreamRequest in the tcp_proxy path, direct access to request_encoder will not be there as we have today.